doc/user_guide/dynamo0p3.rst

```diff
@@ -3048,32 +2983,24 @@ for an ``integer`` exponent.
 inc_X_powreal_a
 ^^^^^^^^^^^^^^^
 
-**inc_X_powreal_a** (*field*, *rscalar*)
+**inc_X_powreal_a** (**field**, *rscalar*)
 
 Raises a field to a ``real`` scalar value and returns the
 field (``X = X**a``)::
 
   field(:) = field(:)**rscalar
 
-where:
-
-* ``type(field_type), intent(in) ::`` **field**
-* ``real(r_def), intent(in) ::`` *rscalar*
-
 inc_X_powint_n
 ^^^^^^^^^^^^^^
 
-**inc_X_powint_n** (*field*, *iscalar*)
+**inc_X_powint_n** (**field**, *iscalar*)
 
 Raises a field to an ``integer`` scalar value and returns
 the field (``X = X**n``)::
 
   field(:) = field(:)**iscalar
 
-where:
-
-* ``type(field_type), intent(in) ::`` **field**
-* ``integer(i_def), intent(in) ::`` *iscalar*
+where ``iscalar`` is an ``integer`` scalar of ``i_def`` precision.
 
 Inner product
 #############
```

Comment:
That's not necessarily correct. Fields can have other precisions. The only supported global sum is ``r_def`` (see #1570) but there's nothing stopping PSyclone from calculating global ``r_def`` sum or inner product of fields with different precision. I specified in the individual built-ins that the resulting scalar must be ``r_def`` (and removed the type definition as requested).

Reply:
But earlier we say that the LFRic infrastructure can only (currently) do global reductions in r_def precision. Is that not true?

Reply:
The global sum arguments as inner product and sum are currently ``r_def``. That doesn't stop the fields in these built-ins being another ``real`` precision, so there will be conversion to ``r_def`` for now. Do you think it would be worth clarifying that in a note?

Reply:
Yes please.

Reply:
I extended the existing notes for inner product and sum built-ins. Feedback is welcome :smiley: I also tidied up some bits and pieces in the section on mixed precision.

Reply:
Looks good - thanks.